### PR TITLE
Date and Date32 type normalization with user's timezone

### LINKED ISSUE #295: toStartOfMonth() Returning Incorrect Values
**What happened**: When using `toStartOfMonth()` in a query in Grafana, the result of the query mimicked what `toEndOfMonth()` would produce with all the values indicating the last day of the month. However, when using the ClickHouse CLI the query returned expected beginning of the month values.

**What you expected to happen**: I would've expected to have the first date of the month returned with a `DateTime` input. For example if entering this: `2022-12-07 02:15:10`, I would expect to be returned this: `2022-12-01`.

**How to reproduce it (as minimally and precisely as possible)**: While connecting Grafana to a ClickHouse database have a field that is a `DateTime` data type. Query the field using the function `toStartOfMonth()`. The result will show the last day of the month of the date that you queried.

An example query:

SELECT toStartOfMonth( "DateTime Field" )
FROM datatable

**Screenshots**

image

**Anything else we need to know?**: When using the clickhouse client to perform the same exact query, the correct result is shown. Screenshot below:

image

**Environment**:

- Grafana version: v8.4.4 (fcfb01faeb)
- Plugin version:
- OS Grafana is installed on: NA
- User OS & Browser: Chrome on Mac OS X 10.15
- Others:

### PR DESCRIPTION
**Pull request by jkaflik**
This resolves #295.

### CODE REVIEW

**Review by asimpson — APPROVED** [review]
LGTM, nice work!

**Comment by jkaflik** [comment]
Hi @asimpson 

PR is ready for review 🚀

File: src/data/CHDatasource.ts [hunk 1]
```diff
@@ -184,11 +186,34 @@ export class Datasource extends DataSourceWithBackend<CHQuery, CHConfig> {
...
+    const targets = request.targets
+      // filters out queries disabled in UI
+      .filter((t) => t.hide !== true)
+      // attach timezone information
+      .map((t) => {
+        return {
+          ...t,
+          meta: {
+            ...t.meta,
+            timezone: this.getTimezone(request),
```
Comment (asimpson) [inline src/data/CHDatasource.ts:209]: tests need to be updated/added for this change.

**Review by asimpson — CHANGES_REQUESTED** [review]

**Comment by asimpson** [comment]
@jkaflik this is looking great, coming together nicely 🏆 ! I left one comment re: testing.

File: src/data/CHDatasource.ts (hunk 1 above)
Comment (jkaflik) [reply src/data/CHDatasource.ts:209]: yeah. let me check that tomorrow

**Review by asimpson — APPROVED** [review]
Nice work!

**Review by asimpson — CHANGES_REQUESTED** [review]
@jkaflik Whoops, I missed this. go tests need to be updated as well. Not sure what the correct values should be given the changes in `clickhouse-go` v.2.7.0.

**Comment by jkaflik** [comment]
@asimpson thanks! pushed changes for backend tests

**Review by asimpson — APPROVED** [review]

**Pull request merged**